app/src/main/java/pm/gnosis/heimdall/ui/safe/recover/extension/ReplaceBrowserExtensionPairingActivity.kt

```diff
@@ -0,0 +1,28 @@
+package pm.gnosis.heimdall.ui.safe.recover.extension
+
+import android.content.Context
+import android.content.Intent
+import pm.gnosis.heimdall.R
+import pm.gnosis.heimdall.ui.safe.pairing.PairingActivity
+import pm.gnosis.heimdall.ui.safe.recover.phrase.ReplaceBrowserExtensionRecoveryPhraseActivity
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asEthereumAddress
+import pm.gnosis.utils.asEthereumAddressString
+
+class ReplaceBrowserExtensionPairingActivity : PairingActivity() {
+    override fun titleRes() = R.string.replace_browser_extension
+
+    override fun onSuccess(extension: Solidity.Address) {
+        val safeAddress = intent.getStringExtra(ReplaceBrowserExtensionPairingActivity.EXTRA_SAFE_ADDRESS).asEthereumAddress()!!
```

Comment:
Why not use the `EXTRA_SAFE_ADDRESS` from this class

Reply:
It's using from the same class. Removed the explicit class reference.